floodviz/tests/test_reference_parser.py

```diff
@@ -0,0 +1,193 @@
+                                           '05484500', '05481300', '05464220', '05458900', '05485605', '05463000',
+                                           '05471200', '05476750', '05411850', '05454220', '05481950', '05416900',
+                                           '05464500', '05487470'],
+                              'display_sites': ['05471200', '05476750', '05411850', '05462000'],
+                              'bbox': [-95.3, 39.8, -91, 43.6], 'start_date': '2008-06-05', 'end_date': '2008-06-15',
+                              'peak_dv_date': '2008-06-09', 'peak_site': '05462000',
+                              'city_geojson_data': {'type': 'FeatureCollection', 'features': [
+                                  {'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': []},
+                                   'properties': {'name': 'Cedar Rapids IA', 'country.etc': 'IA', 'pop': '123243',
+                                                  'capital': '0', 'reftype': 'city'}}]},
+                              'river_geojson_data': '{"type": "FeatureCollection", "features": [{"type": "Feature", "id": 0, "properties": {"gnis_id": " ", "gnis_name": " ", "reftype": "rivers"}, "geometry": {"type": "MultiLineString", "coordinates": []}}]}',
+                              'background_geojson_data': '{"type": "FeatureCollection", "features": [{"type": "Feature", "geometry": {"type": "Polygon", "coordinates": []}, "properties": {"group": "12", "order": " 2951", "region": "illinois", "reftype": "politicalBoundaries"}}]}'}
+
+        self.mock_path = "mock/path.json"
+
+
+    def test_good_data(self):
+        with mock.patch('floodviz.reference_parser.open', mock.mock_open(read_data=self.mock_reference)):
+            parsed_data = reference_parser.parse_reference_data(self.mock_path)
+            self.assertEqual(parsed_data['epsg'], self.mock_response['epsg'])
+            self.assertEqual(parsed_data['display_sites'], self.mock_response['display_sites'])
+            self.assertEqual(parsed_data['bbox'], self.mock_response['bbox'])
+            self.assertEqual(parsed_data['epsg'], self.mock_response['epsg'])
+            self.assertEqual(parsed_data['peak_dv_date'], self.mock_response['peak_dv_date'])
+
+    def test_bad_path(self):
+        parsed_data = reference_parser.parse_reference_data(self.mock_path)
+        self.assertEqual(parsed_data, None)
+
+    def test_missing_field(self):
+        with mock.patch('floodviz.reference_parser.open', mock.mock_open(read_data=self.reference_missing_data)):
+            parsed_data = reference_parser.parse_reference_data(self.mock_path)
+            self.assertEqual(parsed_data, None)
+
+    def test_bad_json(self):
+        with mock.patch('floodviz.reference_parser.open', mock.mock_open(read_data=self.bad_json)):
+            parsed_data = reference_parser.parse_reference_data(self.mock_path)
+            self.assertEqual(parsed_data, None)
+
+
```

Comment:
I'd like to see a separate test for parsing out the reference features into rivers, cities, and background.